chore(td3-54)- Replace H2 database by Postgresql (#87)
talk-to-the-duck/happraisal: docs/use-postgres.md

> # Use postgres and Liquibase

Reviewer: ```suggestion
# Use PostgreSQL and Liquibase
```

Author: Done

## create service in docker compose 

> Objetif: execute locally database

Reviewer: ```suggestion
Objetif: execute the database locally
```

Author: done

https://hub.docker.com/_/postgres

>Note: you can add plugins docker

## Add spring configuration

> Objective: connect application to database

Reviewer: ```suggestion
Objective: connect the application to the database
```

Author: done

```
spring:
  datasource:

>     url: jdbc:postgresql://localhost:5432/happraisal

Reviewer: Could you please change this documentation according to the configuration?

Author: done

    username: happraisameAdimin
    password: happraisalePassword
    driverClassName: org.postgresql.Driver
  jpa:
    open-in-view: false
    generate-ddl: true
    show-sql: true
    database-platform: org.hibernate.dialect.PostgreSQLDialect
    properties:
        hibernate:
            format_sql: true

```

https://docs.liquibase.com/workflows/liquibase-community/existing-project.html
## Use liquibase
Liquibase is an open-source database schema change management solution.

### generate changelog

> liquibase --driver=org.postgresql.Driver --changeLogFile=happraisal.changelog.xml  --classpath=postgresql-9.4-1201-jdbc41.jar --url="jdbc:postgresql://localhost:5432/wms"  --username=<USER_NAME> --password=<PASSWD> --defaultSchemaName=<SCHEMA_NAME> generateChangeLog   

Reviewer: Could you please change this line accordingly to the configuration?
The database version should be 16 or newer.

Author: done

### use liquibase separately
Once the changelog has been generated, we must execute it.

> In a SpringBoot project, it s possible to execute changelog at the startup but this solution should be avoided if horizontal scaling is required.

Reviewer: ```suggestion
In a SpringBoot project, it's possible to execute changelog at the startup but this solution should be avoided if horizontal scaling is required.
```

Author: Done

# Errors
## with docker compose

* After update username and database I got 

``` shell
psql: FATAL:  role "postgres" does not exist
```

> > Soluton: remove volume

Reviewer: ```suggestion
> Solution: remove volume
```

Author: done

# Use PostgreSQL and Liquibase

## create service in docker compose 

> Objetif:  execute the database locally

Reviewer: ```suggestion
Objective:  execute the database locally
```

https://hub.docker.com/_/postgres

>Note: you can add plugins docker

## Add spring configuration

Objective: connect the application to the database
```
spring:
  datasource:
    url: jdbc:postgresql://localhost:5432/happraisal_dbname
    username: happraisal_dbuser
    password: happraisal_dbpassword!
    driverClassName: org.postgresql.Driver
  jpa:
    open-in-view: false
    generate-ddl: true
    show-sql: true
    database-platform: org.hibernate.dialect.PostgreSQLDialect
    properties:
        hibernate:
            format_sql: true

```

https://docs.liquibase.com/workflows/liquibase-community/existing-project.html
## Use liquibase
Liquibase is an open-source database schema change management solution.

### generate changelog

> liquibase --driver=org.postgresql.Driver --changeLogFile=happraisal.changelog.xml  --classpath=postgresql-9.4-1201-jdbc41.jar --url="jdbc:postgresql://localhost:5432/happraisal_dbname"  --username=<USER_NAME> --password=<PASSWD> --defaultSchemaName=<SCHEMA_NAME> generateChangeLog   

Reviewer: Could you please update this version?

### use liquibase separately
Once the changelog has been generated, we must execute it.
In a SpringBoot project, it's possible to execute changelog at the startup but this solution should be avoided if horizontal scaling is required.

# Errors
## with docker compose

> * After update username and database I got 

Reviewer: ```suggestion
* After having updated username and database I got 
```
